<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DebuggerWindow</class>
 <widget class="QDialog" name="DebuggerWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>362</width>
    <height>560</height>
   </rect>
  </property>
  <property name="acceptDrops">
   <bool>false</bool>
  </property>
  <property name="windowTitle">
   <string>Debugger</string>
  </property>
  <property name="toolTip">
   <string/>
  </property>
  <property name="locale">
   <locale language="English" country="UnitedStates"/>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="1" column="0">
    <layout class="QHBoxLayout" name="buttonLayout">
     <item>
      <widget class="QPushButton" name="stepButton">
       <property name="text">
        <string>Step</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="refreshButton">
       <property name="text">
        <string>Refresh</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="0" column="0">
    <layout class="QHBoxLayout" name="contentLayout">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QGroupBox" name="groupBoxMemory">
         <property name="title">
          <string>Memory</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QListWidget" name="listWidgetMemory"/>
          </item>
          <item row="1" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QPushButton" name="loadMemoryButton">
              <property name="text">
               <string>Load Memory</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pcButton">
              <property name="text">
               <string>Jump to PC</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBoxBreakpoints">
         <property name="title">
          <string>Breakpoints</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QLineEdit" name="lineEditBreakpoint">
            <property name="text">
             <string>-1</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <property name="leftMargin">
        <number>20</number>
       </property>
       <item>
        <widget class="QGroupBox" name="groupBoxRegisters">
         <property name="title">
          <string>Registers</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="1" column="0">
           <widget class="QLabel" name="labelBC">
            <property name="text">
             <string>BC:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="lineEditDE">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>0000</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="labelDE">
            <property name="text">
             <string>DE:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="lineEditAF">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>0000</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="lineEditBC">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>0000</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QLineEdit" name="lineEditPC">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>0000</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="7" column="0">
           <widget class="QCheckBox" name="checkBoxSubFlag">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Sub</string>
            </property>
           </widget>
          </item>
          <item row="8" column="0">
           <widget class="QCheckBox" name="checkBoxHalfCarryFlag">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Half</string>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QCheckBox" name="checkBoxZeroFlag">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Zero</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLineEdit" name="lineEditSP">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>0000</string>
            </property>
           </widget>
          </item>
          <item row="9" column="0">
           <widget class="QCheckBox" name="checkBoxCarryFlag">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Carry</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="labelHL">
            <property name="text">
             <string>HL:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLineEdit" name="lineEditHL">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>0000</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="labelPC">
            <property name="text">
             <string>PC:</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QCheckBox" name="checkBoxIME">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>IME</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="labelAF">
            <property name="text">
             <string>AF:</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="labelSP">
            <property name="text">
             <string>SP:</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBoxControls">
         <property name="title">
          <string>Controls</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="1">
           <widget class="QCheckBox" name="checkBoxButtonA">
            <property name="text">
             <string>A</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QCheckBox" name="checkBoxButtonSelect">
            <property name="text">
             <string>Select</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QCheckBox" name="checkBoxButtonStart">
            <property name="text">
             <string>Start</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QCheckBox" name="checkBoxRight">
            <property name="text">
             <string>Right</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QCheckBox" name="checkBoxButtonB">
            <property name="text">
             <string>B</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QCheckBox" name="checkBoxUp">
            <property name="text">
             <string>Up</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QCheckBox" name="checkBoxDown">
            <property name="text">
             <string>Down</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QCheckBox" name="checkBoxLeft">
            <property name="text">
             <string>Left</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>refreshButton</sender>
   <signal>clicked()</signal>
   <receiver>DebuggerWindow</receiver>
   <slot>refresh()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>303</x>
     <y>498</y>
    </hint>
    <hint type="destinationlabel">
     <x>184</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stepButton</sender>
   <signal>clicked()</signal>
   <receiver>DebuggerWindow</receiver>
   <slot>step()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>184</x>
     <y>498</y>
    </hint>
    <hint type="destinationlabel">
     <x>184</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>loadMemoryButton</sender>
   <signal>clicked()</signal>
   <receiver>DebuggerWindow</receiver>
   <slot>loadMemory()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>73</x>
     <y>456</y>
    </hint>
    <hint type="destinationlabel">
     <x>184</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pcButton</sender>
   <signal>clicked()</signal>
   <receiver>DebuggerWindow</receiver>
   <slot>jumpToPC()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>184</x>
     <y>456</y>
    </hint>
    <hint type="destinationlabel">
     <x>184</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxUp</sender>
   <signal>clicked(bool)</signal>
   <receiver>DebuggerWindow</receiver>
   <slot>clickedUp(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>251</x>
     <y>323</y>
    </hint>
    <hint type="destinationlabel">
     <x>179</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxDown</sender>
   <signal>clicked(bool)</signal>
   <receiver>DebuggerWindow</receiver>
   <slot>clickedDown(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>251</x>
     <y>346</y>
    </hint>
    <hint type="destinationlabel">
     <x>179</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxLeft</sender>
   <signal>clicked(bool)</signal>
   <receiver>DebuggerWindow</receiver>
   <slot>clickedLeft(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>251</x>
     <y>369</y>
    </hint>
    <hint type="destinationlabel">
     <x>179</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxRight</sender>
   <signal>clicked(bool)</signal>
   <receiver>DebuggerWindow</receiver>
   <slot>clickedRight(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>251</x>
     <y>392</y>
    </hint>
    <hint type="destinationlabel">
     <x>179</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxButtonA</sender>
   <signal>clicked(bool)</signal>
   <receiver>DebuggerWindow</receiver>
   <slot>clickedA(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>311</x>
     <y>323</y>
    </hint>
    <hint type="destinationlabel">
     <x>179</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxButtonB</sender>
   <signal>clicked(bool)</signal>
   <receiver>DebuggerWindow</receiver>
   <slot>clickedB(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>311</x>
     <y>346</y>
    </hint>
    <hint type="destinationlabel">
     <x>179</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxButtonSelect</sender>
   <signal>clicked(bool)</signal>
   <receiver>DebuggerWindow</receiver>
   <slot>clickedSelect(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>311</x>
     <y>392</y>
    </hint>
    <hint type="destinationlabel">
     <x>179</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxButtonStart</sender>
   <signal>clicked(bool)</signal>
   <receiver>DebuggerWindow</receiver>
   <slot>clickedStart(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>311</x>
     <y>369</y>
    </hint>
    <hint type="destinationlabel">
     <x>179</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEditBreakpoint</sender>
   <signal>returnPressed()</signal>
   <receiver>DebuggerWindow</receiver>
   <slot>breakpointSet()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>98</x>
     <y>497</y>
    </hint>
    <hint type="destinationlabel">
     <x>179</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>refresh()</slot>
  <slot>step()</slot>
  <slot>loadMemory()</slot>
  <slot>jumpToPC()</slot>
  <slot>clickedUp(bool)</slot>
  <slot>clickedDown(bool)</slot>
  <slot>clickedLeft(bool)</slot>
  <slot>clickedRight(bool)</slot>
  <slot>clickedA(bool)</slot>
  <slot>clickedB(bool)</slot>
  <slot>clickedStart(bool)</slot>
  <slot>clickedSelect(bool)</slot>
  <slot>breakpointSet()</slot>
 </slots>
</ui>
